c2corg_ui/static/js/addassociationdirective.js

```diff
@@ -69,7 +79,17 @@ app.AddAssociationController = function(appApi, $attrs) {
  * @export
  */
 app.AddAssociationController.prototype.associate = function(doc) {
-  this.api_.associateDocument(this.parentId, doc).then(function() {
+  var parentId, childId;
+  // we cannot directly associate a wp to route, so we have to
+  // do it the other way around -> route to wp. Same for outing.
+  if ((this.parentDocType === 'routes' && doc['waypoint_type']) || this.parentDocType === 'outings') {
```

Comment:
``this.parentDocType === 'routes' && doc['waypoint_type']``

Is the test about ``doc['waypoint_type']`` a way to check that the "child"  doc is a waypoint? Don't we have a straight way to figure this out?

By the way it's not very clear what is the parent doc and what is the child doc!?
Which one is the current doc and which one is the doc to be associated?